pkg/restapi/vc/operation/operations.go

```diff
@@ -85,30 +110,64 @@ func (c *Operation) createCredentialHandler(rw http.ResponseWriter, req *http.Re
 		return
 	}
 
-	validCredential, err := createCredential(&data)
+	profile, err := c.profileStore.GetProfile(data.Profile)
+	if err != nil {
+		c.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to read profile: %s", err.Error()))
+
+		return
+	}
 
+	validCredential, err := createCredential(profile, &data)
 	if err != nil {
-		c.writeErrorResponse(rw, http.StatusBadRequest, "Failed to write response for create credential failure")
+		c.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to create credential: %s", err.Error()))
+
+		return
+	}
+
+	signedVC, err := c.signCredential(profile, validCredential)
+	if err != nil {
+		c.writeErrorResponse(rw, http.StatusInternalServerError, fmt.Sprintf("failed to sign credential: %s", err.Error()))
 
 		return
 	}
 
 	rw.WriteHeader(http.StatusCreated)
-	c.writeResponse(rw, validCredential)
+	c.writeResponse(rw, signedVC)
+}
+
+func (c *Operation) signCredential(profile *ProfileResponse, vc *verifiable.Credential) (*verifiable.Credential, error) { // nolint:lll
+	signingCtx := &verifiable.LinkedDataProofContext{
+		Creator:       profile.Creator,
+		SignatureType: profile.SignatureType,
+		Suite:         ed25519signature2018.New(),
+		PrivateKey:    c.keySet.private,
+	}
+
+	err := vc.AddLinkedDataProof(signingCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	return vc, nil
 }
 
 func (c *Operation) verifyCredentialHandler(rw http.ResponseWriter, req *http.Request) {
 	body, err := ioutil.ReadAll(req.Body)
 	if err != nil {
-		c.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to read request body: %s", err.Error()))
+		c.writeErrorResponse(rw, http.StatusBadRequest,
+			fmt.Sprintf("failed to read request body: %s", err.Error()))
 
 		return
 	}
 
 	verified := true
 	message := successMsg
 
-	_, _, err = verifiable.NewCredential(body)
+	// Q: should signature suite this be passed in or handled (loaded automatically) by verifiable package
+	// based on signature type in proof
+	// Q: we should have default implementation for key fetcher in verifiable package as well
+	_, _, err = verifiable.NewCredential(body, verifiable.WithEmbeddedSignatureSuites(ed25519signature2018.New()),
```

Comment:
todo: vc DX

eg, 
- one return value. enable a decode pattern for custom structs.
- call this usage parse